src/store/reducers/helper.js

```diff
@@ -74,3 +71,86 @@ export function createComponentMessage(component, props, showAvatar) {
     showAvatar
   });
 }
+
+export const requestSession = (socket) => async function(localID) {
+  return new Promise((resolve, reject) => {
+    socket.emit('session_request', ({ 'session_id': localID }));
+    socket.on('session_confirm', (remoteID) => {
+      console.log(`session_confirm:${socket.id} session_id:${remoteID}`);
+      resolve(remoteID);
+    });
+  })
+}
+
+export function getLocalSession(key) {
+  // Attempt to get local session from sessionStorage
+  const cachedSession = sessionStorage.getItem(key);
+  var session = null;
+  if (cachedSession) {
+    // Found existing session in sessionStorage
+    let parsedSession = JSON.parse(cachedSession)
+    // Format conversation from array of object to immutable Map for use by messages components
+    const formattedConversation = parsedSession.conversation
+      ? Object.values(parsedSession.conversation).map(item => Map(item)).slice()
```

Comment:
I don't think you need a slice here since Map should create a copy of the object to begin with

Reply:
Fixed